src/test_checker/checker_result.rs

```diff
@@ -28,6 +28,22 @@ impl CheckerResult {
             },
         }
     }
+
+    pub fn register_step_error(&mut self, expected: &str, received: &str) {
+        self.register_error(format!(
+            "  expected: {}\n  received: {}\n",
+            expected, received
+        ));
+    }
+
+    pub fn register_error(&mut self, message: String) {
```

Comment:
not &str here?

Reply:
We store that string in `CheckerResult::Failure` here. And I don't want to make that field `&str`, since then that type would require lifetime parameters.